### PR TITLE
gh: dev release workflow

### PR DESCRIPTION
**Pull request by codebien**
Added a workflow for testing the new windows certificate, it's a copy without the uploading phases of the current release process, it builds the request branch, creates the platform binaries and it signs the Windows version.

### CODE REVIEW

File: .github/workflows/dev-release.yml
Comment (na--) [inline .github/workflows/dev-release.yml:90]: This is not the correct legal name, I think

File: .github/workflows/dev-release.yml
Comment (na--) [inline .github/workflows/dev-release.yml:101]: This file doesn't exist :confused: You forgot to rename `build-release.sh`?

Comment (codebien) [reply .github/workflows/dev-release.yml:101]: :face_with_spiral_eyes: this is an error copying from my fork. :man_facepalming:

File: build-release.sh
Comment (na--) [inline build-release.sh:46]: Why this change?

File: .github/workflows/dev-release.yml
Comment (na--) [inline .github/workflows/dev-release.yml:11]: I think you can safely remove `test` from here :thinking: No point in having it just waste time, right?

File: build-release.sh
Comment (na--) [reply build-release.sh]: hmm how about `vX.Y.Z-something`?

File: build-release.sh
Comment (codebien) [inline build-release.sh:12]: This seems something old, is it still required? The `VERSION` value should be always set from the `configure` job.

Comment (codebien) [inline build-release.sh:12]: This seems something old, is it still required? The VERSION value should be always set from the `configure` job and in any case, the tags are not available, the checkout action doesn't fetch them.

File: .github/workflows/dev-release.yml
Comment (codebien) [reply .github/workflows/dev-release.yml:66]: I will change this in a dedicated PR when I will have the right name.

File: build-release.sh
Comment (imiric) [reply build-release.sh:12]: I think it's still valuable if we ever want to run this outside of CI, e.g. locally for some reason.

File: .github/workflows/dev-release.yml
Comment (imiric) [inline .github/workflows/dev-release.yml:102]:
```suggestion
          curl -Lso wix311-binaries.zip https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip
```

We recently had to fix this in https://github.com/grafana/xk6-browser/commit/cbe11c4231a92f32d61d0d8a375d6e7595722fe7, since GitHub added a redirect and the download was failing without `-L`.

File: build-release.sh
Comment (codebien) [reply build-release.sh:12]: Ok, then I think we can keep it with the `if` condition.

Comment (imiric) [reply build-release.sh:12]: Sure, though the `if` is not strictly necessary, since the parameter substitution serves the same purpose if the version is passed as an argument. But it's fine.

**Review by imiric — DISMISSED** [review]
LGTM, but wait for another approval :)

**Comment by codebien** [comment]
@imiric Apparently you moved to `pwsh` before https://github.com/grafana/xk6-browser/commit/36adf8cc32cbd43f6ece520e1aede6db1849e06c, `powershell` seems not to support this config. I can consider migrating entirely to `pwsh` but I would prefer to do it in another PR. I'm going to roll back.

**Comment by imiric** [comment]
Hhmm PowerShell Core worked fine for our scripts, and we could remove the UTF-8 option for the GH env var. What issue are you running into?

**Comment by codebien** [comment]
```
Run curl -Lso wix311-binaries.zip https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip
Invoke-WebRequest : A parameter cannot be found that matches parameter name 'Lso'.
At D:\a\_temp\0c5f23b0-f239-4e54-baae-0d261b5d675a.ps1:2 char:6
+ curl -Lso wix311-binaries.zip https://github.com/wixtoolset/wix3/rele ...
+      ~~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-WebRequest], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```
As reported in private, it requires Powershell Core so I would like to migrate to it, but I would prefer to do it in a dedicated PR.

File: build-release.sh
Comment (na--) [reply build-release.sh:12]: I am confused by this, `${2:-$(git describe --tags --always --dirty)}` means take the second positional argument or use the result of `git describe ...` if that was empty. Why should we also support an external `VERSION` env var, we are only using positional arguments everywhere else in the script :confused: 

Soft :-1: for this, unless we decide to refactor the whole script, but that should happen in some other PR

File: .github/workflows/dev-release.yml
Comment (na--) [reply .github/workflows/dev-release.yml:20]: I see the `workflow_dispatch`, so do we need this job here? Can't you directly use `${{ github.event.inputs.k6_version }}` in the jobs below?

File: .github/workflows/dev-release.yml
Comment (na--) [inline .github/workflows/dev-release.yml:77]: Won't this work? It will remove the need to set `VERSION` here and support it in the built-release script

```suggestion
          ./build-release.sh "dist/" "${{ github.event.inputs.k6_version }}"
```

**Comment by codebien** [comment]
I applied the suggestions and re-tested them, I also added the branch input so we can build eventual PRs not yet merged.

File: .github/workflows/dev-release.yml
Comment (na--) [inline .github/workflows/dev-release.yml:30]: hmm it might be nice if this is a parameter too!

**Review by na-- — DISMISSED** [review]

**Review by na-- — APPROVED** [review]

File: .github/workflows/dev-release.yml
Comment (imiric) [inline .github/workflows/dev-release.yml:68]: I think we need the `-ver-major`, `-ver-minor`, `-ver-patch` flags here, in addition to `-product-version`. Otherwise the version won't be shown properly in the Properties AFAIR. [That's what we do in the main workflow](https://github.com/grafana/k6/blob/59a8883676edf4e28edb39561c33ebd941e05944/.github/workflows/all.yml#L191-L194).

If these are not needed, then the array created on line 54 is not needed either.

**Review by imiric — CHANGES_REQUESTED** [review]

**Review by imiric — APPROVED** [review]
:+1:

**Review by na-- — APPROVED** [review]

**Pull request merged**
